### DISCUSSION TITLE
Need web-service on https://download.mapsforge.org/maps/v5/ to get map files silently

### DISCUSSION THREAD

**Discussion by smartikrosh** [mapsforge/mapsforge#1413] [Q&A] [answered]
Hello,
I wonder if there is a web-service on Mapsforge which allows to download map files silently from an Android app ? Or can it be created..
I mean when a user installs the app, the app downloads a map file (or several files) in the background, without asking a user to visit https://download.mapsforge.org/maps/v5/..... and choose them manually, - just providing his current location and, optionally, a radius. Example:
- current user location is <lat=?, lon=?, radius=?> 
- the app makes a web request to https://download.mapsforge.org/maps/v5/..../?lat=xx&lon=yy&radius=zzz
- the app receives a response as the octet stream containg the required map(s)
- if there are several maps in the server response, they can be merged (the case when a location+radius covers several countries)

Thank you all !

**Comment by devemux86** [comment, marked as answer]
Mapsforge server is a simple download server, there is no extra functionality.

Such tasks are to be handled locally in the applications:

Use the bounds of the countries ([Geofabrik](https://download.geofabrik.de/) poly or other source)
to find where the coordinates are inside and the appropriate map.

Also better not to use the Mapsforge server directly, but the fast mirror.

**Reply by smartikrosh** [reply]
Hi devemux86,
Thank you for your ansver and suggestions.
I found several resources in the Internet which allow to get the bounding boxes at county/region levels. The problem there are no direct relations between regions (or states) there and  on your download server. For example, please look at France - 
https://geodata.ucdavis.edu/gadm/gadm4.1/json/gadm41_FRA_1.json.zip
- actually there are no more regions like "basse-normandie" nor "haute-normandie", only "Normandie", and this is official name in France.
I'm afraid of commiting errors if I'll try to remap myself the coordinates (split/merge etc).
So could you please publish somewhere the actual BoundingBoxes mapping for each item of the directory structure on your server?
For example, France->basse-normandie->BoundingBox.  This info will be relatiely static for long time I think and useful for others.
Thank you!
P.S. Just a remarque - I've noticed that "Crimea" region belongs to Russia in many sources including yours, instead of Ukraine. This could make some people irritated :(

**Reply by devemux86** [reply]
[mapsforge-creator](https://github.com/mapsforge/mapsforge-creator) uses the map regions of Geofabrik: https://download.geofabrik.de

Geofabrik provides .poly files with the extents of the regions,
they can be useful in such calculations.

**Reply by smartikrosh** [reply]
Many thanks!
